### DISCUSSION TITLE
User interface becomes blurry on Wayland with fractional scaling

### DISCUSSION THREAD

**Discussion by duckyondiscord** [ppy/osu#22822] [Q&A]
### Type

Cosmetic

### Bug description

Recently I enabled fractional scaling on my Wayland compositor(Hyprland, based on wlroots). osu does not seem to respect the change and, as Xorg apps running through Xwayland, is blurry.

### Screenshots or videos

Before enabling fractional scaling:
![image](https://user-images.githubusercontent.com/94759537/224029041-75df5e11-d438-441e-967c-8fc19371004f.png)

After enabling fractional scaling:
![image](https://user-images.githubusercontent.com/94759537/224029293-d68f828a-6937-4af3-af21-a91a1d38c3de.png)

### Version

2023.305.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10931613/database.log)
[input.log](https://github.com/ppy/osu/files/10931615/input.log)
[network.log](https://github.com/ppy/osu/files/10931616/network.log)
[performance.log](https://github.com/ppy/osu/files/10931617/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10931618/runtime.log)

**Comment by Bruno5430** [comment]
> as Xorg apps running through Xwayland, is blurry.

Lazer can run on wayland natively without relying on Xwayland, it's just not the default
There were incidents before that were related to Xwayland in lazer https://github.com/ppy/osu/discussions/22763#discussioncomment-5172970 and #22651

**Reply by duckyondiscord** [reply]
Well, `xlsclients` suggests that it is running natively. Regardless, I think it should be the default on Wayland, not to give users a bad experience.
